python_modules/dagster/dagster/_serdes/serdes.py

```diff
@@ -895,6 +926,86 @@ def _pack_value(
     raise SerializationError(f"Unhandled value type {tval}")
 
 
+class _SerdesObjectWrapper(dict):
+    def __init__(self, obj, whitelist_map):
+        self._obj = obj
+        self._whitelist_map = whitelist_map
+        # populate backing native dict to work around c optimizations in json serializer
+        super().__init__({"__serdes": "object_wrapper"})
+
+    def items(self):
+        klass_name = self._obj.__class__.__name__
+        serializer = self._whitelist_map.get_object_serializer(klass_name)
+        yield from serializer.wrap_iter(self._obj, self._whitelist_map)
+
+
+def _wrap_value(
+    val: PackableValue,
+    whitelist_map: WhitelistMap,
+):
+    # this is a hot code path so we handle the common base cases without isinstance
+    tval = type(val)
+    if tval in (int, float, str, bool) or val is None:
+        return cast(JsonSerializableValue, val)
+    if tval is list:
+        return [_wrap_value(item, whitelist_map) for idx, item in enumerate(cast(list, val))]
+    if tval is dict:
+        return {key: _wrap_value(value, whitelist_map) for key, value in cast(dict, val).items()}
```

Comment:
tried custom wrapper objects for list & dict to defer wrapping - was less performant